cmd/osm-bootstrap/osm-bootstrap.go

```diff
@@ -240,19 +245,24 @@ func (b *bootstrap) createDefaultMeshConfig() error {
 }
 
 func (b *bootstrap) ensureMeshConfig() error {
-	_, err := b.meshConfigClient.ConfigV1alpha2().MeshConfigs(b.namespace).Get(context.TODO(), meshConfigName, metav1.GetOptions{})
-	if err == nil {
-		return nil // default meshConfig was found
-	}
-
+	config, err := b.meshConfigClient.ConfigV1alpha2().MeshConfigs(b.namespace).Get(context.TODO(), meshConfigName, metav1.GetOptions{})
 	if apierrors.IsNotFound(err) {
 		// create a default mesh config since it was not found
-		if err = b.createDefaultMeshConfig(); err != nil {
+		return b.createDefaultMeshConfig()
+	} else if err != nil {
+		return err
+	}
+
+	if _, exists := config.Annotations[corev1.LastAppliedConfigAnnotation]; !exists {
```

Comment:
client-go has `Apply()` methods for all the resources it deals with: https://pkg.go.dev/k8s.io/client-go@v0.22.1/kubernetes/typed/core/v1#PodInterface

Is there an easy way for us to add those to our types vs. rolling this by hand?

Reply:
that apply is different somehow.. not quite sure what that's for but I've been playing around with it, and it doesn't add that annotation.

It also requires us hand-rolling those MeshConfigApplyConfiguration types, which would be cumbersome.

Even then, we'd still have to use this method for migrating meshconfig with no annotation to with annotation, as that wouldn't add it.

FWIW this is how kubectl is doing it as well.